### PR TITLE
Display other users' online presence state in dashboard overlay

### PR DESCRIPTION
**Pull request by bdach**
- [x] Depends on https://github.com/ppy/osu/pull/25693
- Server-side PR: https://github.com/ppy/osu-server-spectator/pull/202

https://github.com/ppy/osu/assets/20418176/66de1324-c1d8-495e-b706-9af6f5efc8e9

RFC. General design principles and notes:

- The implementation of this is intentionally jammed into `Metadata{Client,Server}` in order to avoid having a fourth websocket hub connection to manage (fifth overall, if you include chat).
- Clients have to opt into receiving online state updates by calling `IMetadataClient.BeginWatchingUserPresence()`. This was requested by @peppy in IRL discussions.
- I repurposed the "now playing" display for this so that the new one is a superset of the old one. You can still spectate via it, it'll just disable the button on any player that is online but not playing.
- Visual design is very rough and make-work. Most of this is generally a minimal, make-work initial implementation.

### CODE REVIEW

**Comment by cl8n** [comment]
if easily possible, I think loading into a beatmap should leave the status unchanged, rather than switching to "Online" briefly before entering play (0:18 in your video). otherwise cool to see this working :+1:

**Comment by Joehuu** [comment]
> if easily possible, I think loading into a beatmap should leave the status unchanged, rather than switching to "Online" briefly before entering play (0:18 in your video). otherwise cool to see this working 👍

I have an issue opened for that: https://github.com/ppy/osu/issues/21768

File: osu.Game/Online/Metadata/OnlineMetadataClient.cs [hunk 1]
```diff
@@ -116,6 +167,71 @@ public override Task<BeatmapUpdates> GetChangesSince(int queueId)
...
+        public override Task UpdateStatus(UserStatus? status)
+        {
+            if (connector?.IsConnected.Value != true)
+                return Task.FromCanceled(new CancellationToken(true));
+
+            Debug.Assert(connection != null);
+            return connection.InvokeAsync(nameof(IMetadataServer.UpdateStatus), status);
+        }
+
+        public override Task UserPresenceUpdated(int userId, UserPresence? presence)
```
Comment (peppy) [inline osu.Game/Online/Metadata/OnlineMetadataClient.cs:188]: I think this is fine to start with, but we may want to batch this in the future depending on the overhead of individual messages in SignalR. Maybe it's good enough for this kinda throughput though, we'll see!

**Review by peppy — APPROVED** [review]
I don't see anything too offensive 👍 . This is going to be one of those "deploy fast, amend as required" things.

**Pull request merged**